src/app/pages/registration/registration.component.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { RegistrationComponent } from './registration.component';
+import { RegisterFormComponent } from './register-form/register-form.component';
+import { RegisterHeaderComponent } from './register-header/register-header.component';
+import { NgxSpinnerModule, NgxSpinnerService } from 'ngx-spinner';
+import { ReactiveFormsModule, NgForm } from '@angular/forms';
+import { HttpClientModule } from '@angular/common/http';
+import { ToastrModule, ToastrService } from 'ngx-toastr';
+import { RegisterServiceService } from 'src/app/services/register/register-service.service';
+import { User } from 'src/app/models/register/register-details';
+import { HttpTestingController } from '@angular/common/http/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { RegistersuccessComponent } from './registersuccess/registersuccess.component';
+import { registerServiceSpy, resetSpies, toastServiceSpy } from '../../helpers/spies';
+import { of, throwError } from 'rxjs';
+
+
+describe('RegistrationComponent', () => {
+  let component: RegistrationComponent;
+  let fixture: ComponentFixture<RegistrationComponent>;
+
+
+  beforeAll(() => resetSpies([registerServiceSpy]));
+  afterEach(() => resetSpies([registerServiceSpy]));
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ RegistrationComponent,
+        RegisterFormComponent, RegisterHeaderComponent, RegistersuccessComponent
+      ],
+      imports: [
+        NgxSpinnerModule,
+        ReactiveFormsModule,
+        HttpClientModule ,
+        ToastrModule.forRoot(),
+        RouterTestingModule
+    ],
+    providers: [
+      {
+        provide: RegisterServiceService, useValue: registerServiceSpy
+      },
+      {
+        provide: ToastrService, useValue: toastServiceSpy
+      }
+    ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(RegistrationComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('test register user function', () => {
```

Comment:
Hey @akram256, we can make this more descriptive, It's also customary that 'it' is followed by 'should', let us try and make this more descriptive.